torch/nn/modules/module.py

```diff
@@ -763,11 +763,9 @@ def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
                                       .format(key, input_param.shape, param.shape))
                     continue
 
-                if isinstance(input_param, Parameter):
-                    # backwards compatibility for serialized parameters
```

Comment:
do you know what "backwards compatibility for serialized parameters" means here?

Reply:
This comes from a long long time ago [here](https://github.com/pytorch/pytorch/pull/451/files#diff-527b04fec26e044c060ad28b22e842beR240-R242). We used to store `Parameters` in the state_dict. Then we stopped.
Now that we use `no_grad()`. We don't care if it is a Parameter or not. So this is not needed anymore.

Reply:
ah I see, it's just to get the tensor out from the Parameter.

Reply:
Yes.